qrtplib/rtprandom.cpp

```diff
@@ -66,7 +66,7 @@ uint32_t RTPRandom::PickSeed()
 #else
     x += (uint32_t)clock();
 #endif
-    x ^= (uint32_t)((uint8_t *)this - (uint8_t *)0);
+    x ^= (uint32_t)(size_t)this;
```

Comment:
This is substantially different from the original. Did you test it?

Reply:
"rtprandom.cpp:69:37: Performing pointer subtraction with a null pointer may have undefined behavior"

Reply:
Well, for C this is definitely UB. For C++ this should be valid conversion. To make compiler happy i've changed it.
Since this is just a random number generator and converting pointer to integer is valid operation, should be ok.